src/fping.c

```diff
@@ -2539,6 +2564,15 @@ int wait_for_reply(int64_t wait_time)
               }
             }
 
+            if (icmp_request_typ == 13) {
+                if(ip_header_otime_ms != -1 && ip_header_rtime_ms != -1 && ip_header_ttime_ms != -1) {
```

Comment:
I'd think that any 32 bit timestamp value should be accepted. Since ICMP Timestamp support requires raw sockets, `fping` can always read the timestamps, so we do not need to handle the case of not being able to read the data here. I'd prefer to always print the timestamp value.

Reply:
But then I wouldn't need the return `(Timestamp unknown)`, or I would have to check the return to see if `ICMP_TSTAMPREPLY` really happened.
If `ICMP_TSTAMP` is blocked by a firewall, either way only `is unreachable` appears

I don't know if there is a way to answer type 13 without type 14. Because that should be intercepted.

Reply:
An ICMP Timestamp is either answered with an ICMP Timestamp Reply, or possibly an error ICMP message (e.g., ICMP Destination Unreachable). When blocked by a firewall, no reply at all would be common. When no ICMP Timestamp functionality is offered, no reply at all would be expected, as far as I understand it.

Every ICMP Timestamp Reply must include the timestamps, thus I'd say that `(Timestamp unknown)` should never happen and is not really needed. If the packet is too short to contain the timestamps, it should be handled the same way as a too short ICMP Echo reply or other too short messages (see, e.g., lines 2197 and 2309 of `fping.c`).

Reply:
Removed the check